classes/SectionView/SectionView.php

```diff
@@ -97,6 +97,11 @@ protected function display_section_or_speech($args = array()) {
             } else {
                 $URL->insert( array('id'=>$e->getMessage()) );
             }
+            # put the search term back in so highlighting works.
+            # NB: as we don't see the # part of the URL we lose this :(
+            if ( isset($args['s']) && $args['s'] !== '' ) {
```

Comment:
I think going by line #81, `s` will always be set, so this can just be the second half statement.